webscraping_scripts/UmutScrape/format.py

```diff
@@ -0,0 +1,124 @@
+""" 
+1.txt requirements:
+
+
+
+"""
+import re
+import string
+from spellchecker import SpellChecker
+from nltk.corpus import stopwords
+from nltk.tokenize import word_tokenize, sent_tokenize
+import nltk
+nltk.download('stopwords')
+# necessary installs:
+# pip install --user -U nltk
+# pip install pyspellchecker  
+# and the regex library
+
+
+def main():
+    f = open("1.txt", "r")
+    content = f.read()
+    # -------------- getting rid of the increasing numbers in the text ---------------------
+    content_arr = content.split()
+    indexer=[]
+    for ind in range(len(content_arr)):
+        curWord = content_arr[ind]
+        if(curWord.isdigit()):
+            curInt = int(curWord)
+            indexer.append([curInt,ind])
+    collector = []
+    for i in range(len(indexer)):
+        curNum = indexer[i]
+        seqNum = curNum[0]
+        buffer = []
+        if seqNum == -1:
+            continue
+        for j in range(i,len(indexer)):
+            nextNum = indexer[j][0]
+            if nextNum == seqNum + 1:
+                buffer.append([nextNum,indexer[j][1]])
+                indexer[j][0] = -1
+                seqNum = nextNum
+        curNum[0] = -1
+        collector.append(buffer)
+    # get rid of the empty arrays
+    for x in collector:
+        if not x:
+            collector.remove(x)
+    for seq in collector:
+        for ind in seq:
+            curIndex = ind[1]
+            content_arr[curIndex] = -1
+
+    res = [i for i in content_arr if i != -1]
+
+    mystr = ""
+    for ele in res:
+        mystr += ele + " "
+    # -------------- getting rid of the increasing numbers in the text ends here ---------------------
+
+
+    # ----------- spellchecker - optional -----------
+    spelling = SpellChecker()
+    def spelling_checks(text):
+        correct_result = []
+        typo_words = spelling.unknown(text.split())
+        for word in text.split():
+            if word in typo_words:
+                correct_result.append(spelling.correction(word))
+            else:
+                correct_result.append(word)
+        return " ".join(correct_result)
+
+    mystr = spelling_checks(mystr)
+    # -------------- end of spellchecker --------------
```

Comment:
I don't think a spellchecker should be necessary considering these are policies that have probably undergone several reviews and edits. Have you seen any misspelled words?

Reply:
No, there weren't any misspelled words, I wanted to include a spellchecker because in the future there might be data that needs to be fixed.